### PR TITLE
Kill container directly if timeout is 0.

### PR DESCRIPTION
**Pull request by Random-Liu**
Based on [CRI Defination](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto#L652-L654), timeout 0 means `forcibly terminate the container immediately`.

So skip graceful deletion when timeout is 0.

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

File: pkg/server/container_stop.go [hunk 1]
```diff
@@ -65,22 +65,24 @@ func (c *criContainerdService) StopContainer(ctx context.Context, r *runtime.Sto
...
-	// TODO(random-liu): [P1] Get stop signal from image config.
-	stopSignal := unix.SIGTERM
-	glog.V(2).Infof("Stop container %q with signal %v", id, stopSignal)
-	_, err = c.containerService.Kill(ctx, &execution.KillRequest{ID: id, Signal: uint32(stopSignal)})
-	if err != nil {
-		if isContainerdContainerNotExistError(err) {
-			return &runtime.StopContainerResponse{}, nil
+	if r.GetTimeout() > 0 {
+		// TODO(random-liu): [P1] Get stop signal from image config.
+		stopSignal := unix.SIGTERM
```
Comment (mikebrow) [inline pkg/server/container_stop.go]: did we want to send a SIGKILL if the SIGTERM times out?

Comment (Random-Liu) [inline pkg/server/container_stop.go]: This means that if a timeout is set, we use `SIGTERM` to gracefully stop the container;
If not, we `SIGKILL` the container directly.

Comment (mikebrow) [inline pkg/server/container_stop.go]: if zero no kill?

Comment (Random-Liu) [inline pkg/server/container_stop.go]: `Delete` includes a `SIGKILL`.

Comment (mikebrow) [inline pkg/server/container_stop.go]: Please comment below that calling c.containerService.Delete results in a SIGKILL being sent

Comment (Random-Liu) [inline pkg/server/container_stop.go]: Will do.

Comment (mikebrow) [inline pkg/server/container_stop.go]: calling delete on a running container will not kill.. so probably need to send a kill if timeout is zero...

Comment (Random-Liu) [inline pkg/server/container_stop.go]: Done.

**Comment by Random-Liu** [comment]
Apply LGTM based on https://github.com/kubernetes-incubator/cri-containerd/pull/75#pullrequestreview-43831790.

**Comment by Random-Liu** [comment]
Offline discussed with @yujuhong, she's also ok with the change. Merge it.

**Pull request merged**